src/main/java/org/gitlab4j/api/ImportExportApi.java

```diff
@@ -108,18 +125,32 @@ public File downloadExport(Object projectIdOrPath, File directory) throws GitLab
             if (directory == null)
                 directory = new File(System.getProperty("java.io.tmpdir"));
 
-            String disposition = response.getHeaderString("Content-Disposition");
-            String filename;
-            if(disposition == null) {
-                // On GitLab.com the Content-Disposition returned is null
-                if(projectIdOrPath instanceof Project) {
-                    filename = ((Project) projectIdOrPath).getPath();
+            if(filename == null) {
+                // No filename provided
+                String disposition = response.getHeaderString("Content-Disposition");
+                if (disposition == null) {
+                    // On GitLab.com the Content-Disposition returned is null
+                    if (projectIdOrPath instanceof Project) {
+                        String template = "%1$tY-%1$tm-%1$td_%1$tH-%1$tM-%1$tS_%2$s_export.tar.gz";
+                        filename = String.format(template, new Date(), ((Project) projectIdOrPath).getPathWithNamespace().replace('/', '_'));
+                        // filename = "2019-06-10_10-28-52_namespace-group_test-project_export.tar.gz"
+                    } else if(projectIdOrPath instanceof String) {
+                        String template = "%1$tY-%1$tm-%1$td_%1$tH-%1$tM-%1$tS_%2$s_export.tar.gz";
+                        filename = String.format(template, new Date(), projectIdOrPath);
+                        // filename = "2019-06-10_10-28-52_test-project_export.tar.gz"
+                    } else if(projectIdOrPath instanceof Integer) {
+                        String template = "%1$tY-%1$tm-%1$td_%1$tH-%1$tM-%1%tS_projectid-%2$d_export.tar.gz";
+                        filename = String.format(template, new Date(), projectIdOrPath);
+                        // filename = "2019-06-10_10-28-52_projectid_3115610_export.tar.gz"
+                    } else {
+                        // Fallback for strange types
```

Comment:
This will never occur as the call to ```getProjectIdOrPath(projectIdOrPath)``` above will throw an exception if it is not a valid Project instance,  project path, or project ID (Integer).